HIP/hip-991.md

```diff
@@ -44,182 +44,310 @@ This document outlines the development of a fixed fee system for the submission
+**User Personas:**
 
-### **Protobuf Messages Needed**
+* **Topic Operator:** Entities or individuals who manage and operate topics, interested in stable revenue and easy management of topic-related transactions.
+* **Topic Operator:** Entities or individuals who manage and operate topics, interested in reducing the noise of public topic IDs/making it more difficult to spam an open topic.
+* **Regular User:** End users who interact with topics, adding paid methods of interaction and submission to topics.
 
-### **FixedFee**
+**User Stories:**
 
-The `FixedFee` message represents a fixed fee charged for each message submission to a topic. This protobuf message already exists so it can be re-used.
+* As a **Topic Operator**, I want to set a fixed fee for submitting messages to my topic, so that I can ensure a consistent revenue stream and manage my topic effectively.
+* As a **Regular User**, I want to know the exact cost of interacting with a topic.
 
-```
-message FixedFee {
-  // The number of units to assess as a fee
-  int64 amount = 1;
-  // The denomination of the fee; taken as hbar if left unset
-  TokenID denominating_token_id = 2;
-}
-```
+## Specification
 
-### **CustomFixedFee**
+### Overview
 
-The `CustomFixedFee` message defines the type of fee and the account receiving the fee assessed during a message submission to the associated topic. A custom fee may only be fixed and must specify a fee collector account to receive the assessed fees.
+We propose adding a fixed fee mechanism to the Hedera Consensus Service (HCS) for topic messages, similar to the custom fee structures in the Hedera Token Service (HTS). This will involve creating new protocol buffers (protobuf) messages and modifying existing ones to accommodate fixed fees for topics.
 
-```
-message CustomFixedFee {
-  // Fixed fee to be charged
-  FixedFee fixed_fee = 1;
+### Requirements and Scope
 
-  // The account to receive the custom fee
-  AccountID fee_collector_account_id = 3;
-}
-```
+* Topics can have a Fee Schedule Key set at creation time.
+* Fee Schedule Key can manage fee updates and send free messages to the topic.
+* The Fee Schedule Key can be updated with an update topic transaction signed by the Admin Key and the new Fee Schedule Key.
+* If the topic was created without a Fee Schedule Key, the key cannot be added later.
+* Topics can have fees for submitting messages.
+* Fees can be set at creation time.
+* Fees can be changed (updated or removed) from a topic with an update topic transaction signed by the Fee Schedule Key.
+* Fees are defined as a list of custom fee.
+* A custom fee can be set in HBAR or HTS fungible tokens, and must have an accountID as a receiver.
+* Fee payer must set an allowance in HBAR or HTS tokens to be used to pay fees for message submissions, on an HCS topic basis.
```

Comment:
These fees are per message allowance and not total allowance. Don't know if we need to be overly specific here but maybe that's an important callout as it's different than allowances today.

Reply:
Side note: in case you don't care about an overall max, you are still forced to use a very high max total allowance like users do now when they don't care about a real limit. The other option would have been to introduce something like a negative number for infinite, but it would have been an additional complexity to implement and explain, and a different behaviour vs the common practice.

Reply:
Knowing it can be executed in the same transaction is very helpful. 

If we make sure that default value is auto-injected in the SDKs if it's not defined then I'm ok with it. I think there's a lot of use cases where people will not want to set a max allowance.

Reply:
Yep, that can be achieved by the developers because they can set that default values as they prefer for their users.

But the SDK always starts with the safest default values for any parameters (so in this case 0 max allowance, 0 max allowance per message, no owner, no spender) so developers don't unintentionally harm their users.